nexus/src/lib.rs

```diff
@@ -164,11 +164,15 @@ impl Server {
  * Run an instance of the [Server].
  */
 pub async fn run_server(config: &Config) -> Result<(), String> {
-    let log = config
-        .log
-        .to_logger("nexus")
-        .map_err(|message| format!("initializing logger: {}", message))?;
-    if let Err(e) = usdt::register_probes() {
+    use slog::Drain;
+    let (drain, registration) = slog_dtrace::with_drain(
+        config
```

Comment:
(No action)

This is kinda funny, the tuple looks like it's basically acting as a `Result` type. Like, the only point of "registration" is to propagate an error if we fail, right? Wouldn't it be more idiomatic to return `Result<Drain>`?

Reply:
Yeah, I get why it seems odd :) I wanted to return _both_ values regardless of whether registration failed, rather than one or the other. Since `slog-dtrace` is a pretty general-purpose library, we thought it'd be best to leave it up to the consumer to decide how to handle registration failures. So we hand back both the new `Drain` and the result of registration -- callers can decide to panic or return their own `Result`.

Reply:
But isn't the registration object exclusively:

- Success (no value)
- Error (error value)

? What would a caller do with the registration object that would be different from interacting with a `Result`?

Reply:
They could ignore the error and still use the drain. For example, the `slog_dtrace::with_drain` function duplicates log messages to the provided `Drain` and a new `Dtrace` object. Callers could decide that they want to continue their application, only logging to the first drain and ignoring the fact that those messages won't show up via DTrace.